src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+    using SafeMathUint256 for uint256;
+    using SafeMathInt256 for int256;
+
+    enum ReportingState {
+        AUTOMATED,
+        LIMITED,
+        ALL,
+        FORK,
+        FINALIZED
+    }
+
+    ReportingState private reportingState = ReportingState.AUTOMATED;
+
+    // CONSIDER: change the payoutNumerator/payoutDenominator to use fixed point numbers instead of integers; PRO: some people find fixed point decimal values easier to grok; CON: rounding errors can occur and it is easier to screw up the math if you don't handle fixed point values correctly
+    uint256 private payoutDenominator;
+    uint256 private feePerEthInAttoeth;
+
+    // CONSIDER: we really don't need these
+    int256 private maxDisplayPrice;
+    int256 private minDisplayPrice;
+
+    // CONSIDER: figure out approprate values for these
+    uint256 private constant AUTOMATED_REPORTER_DISPUTE_BOND_AMOUNT = 11 * 10**20;
+    uint256 private constant LIMITED_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**21;
+    uint256 private constant ALL_REPORTERS_DISPUTE_BOND_AMOUNT = 11 * 10**22;
+
+    uint256 private constant MIN_NUM_OUTCOMES = 2;
+    uint256 private constant MIN_PAYOUT_DENOMINATOR = 2;
+    uint256 private constant MAX_FEE_PER_ETH_IN_ATTOETH = 5 * 10 ** 17;
+    int256 private constant DISPLAY_PRICE_MIN = -(2 ** 254);
+    int256 private constant DISPLAY_PRICE_MAX = 2 ** 254;
+    uint256 private constant SET_COST_MULTIPLIER_MAX = 2 ** 254;
+    uint256 private constant APPROVAL_AMOUNT = 2 ** 254;
+    uint256 private constant AUTOMATED_REPORTING_DURATION_SECONDS = 3 days;
+    uint256 private constant AUTOMATED_REPORTING_DISPUTE_DURATION_SECONDS = 3 days;
+    address private constant NULL_ADDRESS = address(0);
+
+    ReportingWindow private reportingWindow;
+    uint256 private endTime;
+    uint8 private numOutcomes;
+    uint256 private marketCreationBlock;
+    bytes32 private topic;
+    address private automatedReporterAddress;
+    mapping(bytes32 => ReportingToken) private reportingTokens;
+    Cash private denominationToken;
+    address private creator;
+    IShareToken[] private shareTokens;
+    uint256 private finalizationTime;
+    bool private automatedReportReceived;
+    bytes32 private tentativeWinningPayoutDistributionHash;
+    bytes32 private finalPayoutDistributionHash;
+    DisputeBondToken private automatedReporterDisputeBondToken;
+    DisputeBondToken private limitedReportersDisputeBondToken;
+    DisputeBondToken private allReportersDisputeBondToken;
+    uint256 private validityBondAttoeth;
+    uint256 private automatedReporterBondAttoeth;
+
+    function initialize(ReportingWindow _reportingWindow, uint256 _endTime, uint8 _numOutcomes, uint256 _payoutDenominator, uint256 _feePerEthInAttoeth, Cash _denominationToken, address _creator, int256 _minDisplayPrice, int256 _maxDisplayPrice, address _automatedReporterAddress, bytes32 _topic) public payable beforeInitialized returns (bool _success) {
```

Comment:
Since this contract is `initializable` I wonder if we should tag all of the methods in here with `afterInitialized` (or whatever the modifier is called).  I'm not sure yet how I feel about the practice yet, it is nice for small contracts that have a substantial amount of code executed during/before initialization, but in our case I don't think there is any pre-initialization code.